packages/plugin-ext/src/plugin/output-channel/log-output-channel.ts

```diff
@@ -0,0 +1,132 @@
+// *****************************************************************************
+// Copyright (C) 2023 STMicroelectronics and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+// *****************************************************************************
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+import { Emitter } from '@theia/core/shared/vscode-languageserver-protocol';
+import * as theia from '@theia/plugin';
+
+import { OutputChannelRegistryMain, PluginInfo } from '../../common/plugin-api-rpc';
+import { OutputChannelImpl } from './output-channel-item';
+import { LogLevel } from '../types-impl';
+
+export class LogOutputChannelImpl extends OutputChannelImpl implements theia.LogOutputChannel {
+
+    readonly onDidChangeLogLevelEmitter: Emitter<theia.LogLevel> = new Emitter<theia.LogLevel>();
+    readonly onDidChangeLogLevel: theia.Event<theia.LogLevel> = this.onDidChangeLogLevelEmitter.event;
+    public logLevel: theia.LogLevel;
+
+    constructor(override readonly name: string, protected override readonly proxy: OutputChannelRegistryMain, protected override readonly pluginInfo: PluginInfo) {
+        super(name, proxy, pluginInfo);
+        this.setLogLevel(LogLevel.Info);
+    }
+
+    setLogLevel(level: theia.LogLevel): void {
+        if (this.logLevel !== level) {
+            this.logLevel = level;
+            this.onDidChangeLogLevelEmitter.fire(this.logLevel);
+        }
+    }
+
+    getLogLevel(): theia.LogLevel {
+        return this.logLevel;
+    }
+
+    override append(value: string): void {
+        this.info(value);
```

Comment:
Yes it does not explicitely say that, but if we do use the basic append mechanism, the output would look like this:

```
2023-04-21T08:13:28.630Z [Warning] Logging a WARNING message via LogOutputChannel
2023-04-21T08:13:33.981Z [Error] Logging an ERROR message via LogOutputChannel
Append text content as new line to LogOutputChannel
2023-04-21T08:13:41.973Z [Warning] Logging a WARNING message via LogOutputChannel
Append text content to LogOutputChannel2023-04-21T08:13:49.676Z [Warning] Logging a WARNING message via LogOutputChannel
```

Using `.info()` ensures that all logged output is formatted consistently.
VS Code has implemented it the same way, which I would prefer as well as the log output is overall more consistent.
But if you insist, we can also change it to the simple append mechanism.

Reply:
Hmh...that's odd, why have `append` if it's exactly like `info`? But since this is VS Code API, compatibility trumps common sense. I'd leave it like it is.